### ISSUE TITLE
structure reference issue

### ISSUE THREAD

**Issue by LADSoft** [LADSoft/OrangeC#374]
consider:

```
struct sym
{
...
};
typedef struct sym SYMBOL;

int myFunc(SYMBOL *sym)
{
      SYMBOL *p = (struct sym *)malloc(sizeof(SYMBOL));      
}
```

the cast on the malloc() line fails because the compiler finds the function argument rather than the structure name.   This seems to be an existing issue so I deferred fixing it...

**Comment by LADSoft** [comment]
related issue:

> struct sym
> {
>      struct mystruct
>      {
>              ...
>      };
> ...
> };
> typedef struct sym SYMBOL;
> 
> int myFunc(SYMBOL *sym)
> {
>       auto aa = (sym::mystruct*)malloc(sizeof(sym::mystruct));
> }

**Comment by LADSoft** [comment]
this seems fixed in version 6.70 of the compiler...
